### PR TITLE
Augment Getting Help docs.

### PR DESCRIPTION
**Pull request by benjyw**
Lay out expectations for how to ask for help.

### CODE REVIEW

File: src/pages/community/getting-help.mdx [hunk 1]
```diff
@@ -6,33 +6,61 @@ How to engage with the Pants community.
...
+  The README.md in this repo should provide clear instructions on how to reproduce the problem, and the architecture and OS on which it can be reproduced.
+  Posting code and config snippets to Slack is not a good alternative, as we won't know if they contain enough information to reproduce the problem.
+  A self-contained repo is far more useful.
+- Then [file an issue](https://github.com/pantsbuild/pants/issues/new/choose) that states the problem and the steps you've taken to attempt to solve it, and a link to the reproduction repo. If you find
+  an existing issue that seems relevant, please link to it, or post directly to that existing issue if you're reasonably sure it describes the same problem.
+- Then, feel free to reach out in the #general channel in Slack with a link to your issue, to kickstart the discussion.
+
+Keep in mind that this community support is provided _voluntarily_. When you ask for help you're asking busy people to devote their valuable time to solving your problem.
+You are far more likely to get a positive response if you demonstrate that you've put in the basic legwork first.
```
Comment (huonw) [inline src/pages/community/getting-help.mdx:27]: I strongly appreciate encouraging people to provide the right info straight up, but I'm a little concerned about phrasing some of this stuff too strongly in terms of "you must do all these things [or else you'll be ignored for process reasons]" which seems... scary, especially for someone new to open source.

I'd personally prefer to phrase it in terms of what's in it for the asker, i.e. focus on "help us help you" / "you'll fix your problem quicker if you make it easy for us to help".

For instance:

```suggestion
You are far more likely to get a positive response if you make it easy for people to help you.
```

But I think potentially this whole section could be reframed slightly to be cast in that light, especially the list above.

What do you think?

File: src/pages/community/getting-help.mdx [hunk 2]
```diff
@@ -6,33 +6,61 @@ How to engage with the Pants community.
...
-## Slack
+## How to get help
 
-The best place to become acquainted with Pants community members, exchange informal chat and ask quick questions is at the project's Slack. You can join using [this link](https://docs.google.com/forms/d/e/1FAIpQLSf9zgf-MXRnVDJbrVEST3urqneq7LCcy0zw8qU-GH4hPMn52A/viewform?usp=sf_link).
+Pants community members will do their best to answer any questions that are posed in a respectful way. General questions about Pants,
+its features, language support, roadmap etc. are always welcome.
+
+If you need help debugging a specific problem you're encountering in your repo, then we ask that you help us to help you by adhering to the following:
+
+- First, do your best to solve the issue on your own. The Pants [source code](https://github.com/pantsbuild/pants), [issue tracker](https://github.com/pantsbuild/pants/issues), and [documentation](https://pantsbuild.org)
```
Comment (huonw) [inline src/pages/community/getting-help.mdx:16]: Minor thing, but maybe this should be a numbered list:

```suggestion
1. First, do your best to solve the issue on your own. The Pants [source code](https://github.com/pantsbuild/pants), [issue tracker](https://github.com/pantsbuild/pants/issues), and [documentation](https://pantsbuild.org)
```

File: src/pages/community/getting-help.mdx [hunk 3]
```diff
@@ -6,33 +6,61 @@ How to engage with the Pants community.
...
+  Posting code and config snippets to Slack is not a good alternative, as we won't know if they contain enough information to reproduce the problem.
+  A self-contained repo is far more useful.
+- Then [file an issue](https://github.com/pantsbuild/pants/issues/new/choose) that states the problem and the steps you've taken to attempt to solve it, and a link to the reproduction repo. If you find
+  an existing issue that seems relevant, please link to it, or post directly to that existing issue if you're reasonably sure it describes the same problem.
+- Then, feel free to reach out in the #general channel in Slack with a link to your issue, to kickstart the discussion.
+
+Keep in mind that this community support is provided _voluntarily_. When you ask for help you're asking busy people to devote their valuable time to solving your problem.
+You are far more likely to get a positive response if you demonstrate that you've put in the basic legwork first.
+
+As an open-source project, we always appreciate a willingness to dive in and fix issues yourself, with our guidance.
```
Comment (huonw) [inline src/pages/community/getting-help.mdx:29]: We could potentially also call out "provide support to others" as a useful activity, in addition to fixing issues.

File: src/pages/community/getting-help.mdx [hunk 4]
```diff
@@ -6,33 +6,61 @@ How to engage with the Pants community.
...
 
-The best place to become acquainted with Pants community members, exchange informal chat and ask quick questions is at the project's Slack. You can join using [this link](https://docs.google.com/forms/d/e/1FAIpQLSf9zgf-MXRnVDJbrVEST3urqneq7LCcy0zw8qU-GH4hPMn52A/viewform?usp=sf_link).
+Pants community members will do their best to answer any questions that are posed in a respectful way. General questions about Pants,
+its features, language support, roadmap etc. are always welcome.
+
+If you need help debugging a specific problem you're encountering in your repo, then we ask that you help us to help you by adhering to the following:
+
+- First, do your best to solve the issue on your own. The Pants [source code](https://github.com/pantsbuild/pants), [issue tracker](https://github.com/pantsbuild/pants/issues), and [documentation](https://pantsbuild.org)
+  are useful resources for this.
+- If you're still stuck, and would like a Pants contributor to look at your problem, please provide a small public GitHub repo that reproduces the problem.
```
Comment (huonw) [inline src/pages/community/getting-help.mdx:18]: In addition to the framing general concern, sometimes it's quite hard to create a reproducer, e.g. inconsistent things like https://github.com/pantsbuild/pants/issues/19748, or performance issues at scale. Obviously a reproducer is helpful and really valuable, but it'd be unfortunate to completely discourage people from filing issues about those sort of things, at least as a place to have the community start "taking notes"/collaborating a reproducer.

I don't think we're at the scale of incoming issues where it's a major burden to encourage/remind users for a reproducer.

So suggestion for two actions:

- the phrasing of this to be something along the lines "provide a reproducer, or if not possible, all the information you can, e.g. `pants.toml` and relevant `BUILD` files"
- we (i.e. Pants maintainers) could start labelling such issues "requires reproducer" or something and then semi-regularly do a (manual) clean-up of such issues? E.g. if there's a bunch of unreproduced issues sitting around for a while, go through and close them.

Thoughts?

**Review by huonw — COMMENTED** [review]
I like this in principle, thanks for taking it on

File: src/pages/community/getting-help.mdx (hunk 4 above)
Comment (sureshjoshi) [reply src/pages/community/getting-help.mdx:18]: > the phrasing of this to be something along the lines "provide a reproducer, or if not possible, all the information you can, e.g. pants.toml and relevant BUILD files"

What about some location (ticket, repo, whatever) where we have a set of common things we ask for (maybe something we could automate, anonymized into Pants CLI) - so, pants.toml, BUILD files, logs, neofetch data, etc... There would end up being a ton of superfluous data 🤷🏽 

> we (i.e. Pants maintainers) could start labelling such issues "requires reproducer" or something and then semi-regularly do a (manual) clean-up of such issues? E.g. if there's a bunch of unreproduced issues sitting around for a while, go through and close them.

OfficeJ auto-closes in 4 days, which is infuriating: Screenshot 2024-03-07 at 10 01 38

After an initial human triage of staleness, I'd be more open to a longer-running auto-close bot at like 3 months or something? I was looking at the issue backlog and seeing tickets that haven't been updated from mid-2018

File: src/pages/community/getting-help.mdx (hunk 1 above)
Comment (sureshjoshi) [reply src/pages/community/getting-help.mdx:27]: Are there any projects that do this kind of language well (friendly, while direct and helpful?).

I'm always a fan of how Svelte phrases things, for example.

Screenshot 2024-03-07 at 10 09 38

File: src/pages/community/getting-help.mdx (hunk 4 above)
Comment (huonw) [reply src/pages/community/getting-help.mdx:18]: > What about some location (ticket, repo, whatever) where we have a set of common things we ask for (maybe something we could automate, anonymized into Pants CLI) - so, pants.toml, BUILD files, logs, neofetch data, etc... There would end up being a ton of superfluous data 🤷🏽

Putting it in a specific list of information to these docs might be a start?

> OfficeJ auto-closes in 4 days, which is infuriating: 
>
> After an initial human triage of staleness, I'd be more open to a longer-running auto-close bot at like 3 months or something? I was looking at the issue backlog and seeing tickets that haven't been updated from mid-2018

Yeah, fully agreed: auto-closers are user-hostile. I'm thinking the label could be used for a human to go and find issues easily, and then close them out manually (or potentially not close them, e.g. if someone did provide a reproducer but it wasn't noticed). Pants doesn't receive that many issues 😄

File: src/pages/community/getting-help.mdx (hunk 1 above)
Comment (huonw) [reply src/pages/community/getting-help.mdx:27]: > I'm always a fan of how Svelte phrases things, for example.

Ah yeah, I like it too.

Somewhat tangential but that text there makes me wonder if we could also update our bug report template to be more specific about what's expected.

>  Are there any projects that do this kind of language well (friendly, while direct and helpful?).

Good question! I don't know of any particularly good examples of the top of my head.


I know Rust (used to) put a lot of effort into helping new-comers, and this is their current "generic bug" issue template:

``````markdown
<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
<code>
```

I expected to see this happen: *explanation*

Instead, this happened: *explanation*

### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
<version>
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>
``````

File: src/pages/community/getting-help.mdx (hunk 4 above)
Comment (benjyw) [reply src/pages/community/getting-help.mdx:18]: > I don't think we're at the scale of incoming issues where it's a major burden to encourage/remind users for a reproducer.

Not sure about that. It is a frequent nuisance when people post questions lacking important information, or dump snippets directly into slack.

File: src/pages/community/getting-help.mdx (hunk 1 above)
Comment (benjyw) [reply src/pages/community/getting-help.mdx:27]: I'll just take this bit out.

**Comment by benjyw** [comment]
PTAL, thanks!

**Comment by tgolsson** [comment]
Still think it's a bit harshly worded but I'm struggling to find alternatives. 

> Many issues, big or small, can be solved on your own or may have existing documentation. It is always worth checking the issue tracker, <...>. Even if you can't reach a complete solution, doing this groundwork is helpful and gives us [the community??] equal investment in helping you.

Maybe a bit wordy, and the end seems a bit robotic... not sure.

**Comment by benjyw** [comment]
Which part specifically seems too harsh? The "first try and solve it on your own" part?

**Review by cognifloyd — COMMENTED** [review]
I think this might be a way to soften things.

**Comment by tgolsson** [comment]
> Which part specifically seems too harsh? The "first try and solve it on your own" part?

Yes. I think the rest is good. Just this first sentence feels very condescending. I think it has this implicit assumption that the user is doesn't try on their own without prompting. Like a quiz for school kids. "Give each problem a try before raising your hand".

File: src/pages/community/getting-help.mdx [hunk 5]
```diff
@@ -6,33 +6,59 @@ How to engage with the Pants community.
...
 
-The best place to become acquainted with Pants community members, exchange informal chat and ask quick questions is at the project's Slack. You can join using [this link](https://docs.google.com/forms/d/e/1FAIpQLSf9zgf-MXRnVDJbrVEST3urqneq7LCcy0zw8qU-GH4hPMn52A/viewform?usp=sf_link).
+Pants community members will do their best to answer any respectfully posed questions. General questions about Pants,
+its features, language support, roadmap etc. are always welcome!
 
-We encourage you to introduce yourself in the `#welcome` channel!
+If you need help debugging a specific problem you're encountering in your repo, then we ask that you help us to help you:
+
+1. First, do your best to solve the issue on your own. The Pants [source code](https://github.com/pantsbuild/pants), [issue tracker](https://github.com/pantsbuild/pants/issues), and [documentation](https://pantsbuild.org)
+   are useful resources for this.
```
[the remark this answers is not in the document]
Comment (sureshjoshi) [reply src/pages/community/getting-help.mdx:17]: 👍🏽

**Comment by sureshjoshi** [comment]
> > Which part specifically seems too harsh? The "first try and solve it on your own" part?
> 
> Yes. I think the rest is good. Just this first sentence feels very condescending. I think it has this implicit assumption that the user doesn't try on their own without prompting. Like a quiz for school kids. "Give each problem a try before raising your hand".

I think Jacob's variant mitigates this well enough to put it up. 

> implicit assumption that the user doesn't try on their own without prompting

I think based on history, that does happen often though 🤷🏽 

Calls to action work. About half of my feedback on Slack these past years has been a link to the docs, or a link to a prior Slack conversation - and that's basically it.

The people who would already dig through resources before trying to solve problems themselves will continue to do so, and those are are a bit less familiar with the codebase (or maybe development in general) can get a short list of good resources to take a crack at first.

And as lowly as I think of the current state/hype of and around "AI", this would be a picture-perfect use case for LLMs. Soak up all the docs and Slack history, weight it by recency and then allow users to type in a problem and point them to possible doc pages or Slack convos 🤌 

In the meantime, I think a small, polite call to action + resources is pretty good.

**Review by cognifloyd — APPROVED** [review]

**Review by tgolsson — APPROVED** [review]
